library/base/src/main/kotlin/org/cru/godtools/base/Settings.kt

```diff
@@ -157,6 +167,40 @@ class Settings internal constructor(private val context: Context, coroutineScope
     }
     // endregion Dashboard Settings
 
+    // region optInNotification
+    private val dateFormat = SimpleDateFormat("MM/dd/yyyy", Locale.getDefault())
+
+    fun getLastPromptedOptInNotification(): Date? {
+        val lastPrompted = prefs.getString(LAST_PROMPTED_OPT_IN_NOTIFICATION, null)
+
+        return lastPrompted?.let {
+            try {
+                dateFormat.parse(it)
+            } catch (e: ParseException) {
+                null
+            }
+        }
+    }
+
+    fun getOptInNotificationPromptCount(): Int {
+        val promptCount = prefs.getInt(OPT_IN_NOTIFICATION_PROMPT_COUNT, 0)
+
+        return promptCount
+    }
+
+    fun recordOptInNotificationPrompt() {
+        val currentPromptCount = getOptInNotificationPromptCount()
+        val updatedPromptCount = currentPromptCount + 1
+
+        val dateString = dateFormat.format(Date())
+
+        prefs.edit {
+            putString(LAST_PROMPTED_OPT_IN_NOTIFICATION, dateString)
```

Comment:
Let's store the last prompted timestamp as the time millis, that way you don't have to format & parse the date when loading it

Reply:
My own personal preference is to use the java8 time APIs instead of the older `Date` class. They are much more full featured.

Since you are storing just a date, there is a `LocalDate` object that can be used, it provides `LocalDate.toEpochDay()` and `LocalDate.ofEpochDay()` which can be used for storing the value in preferences. It also supports comparison by doing something like: `storedDate.plusDays(5) < LocalDate.now()`

It looks like we have already enabled android desugaring for this library so we would have access to those APIs without any issues